packages/flutter/lib/src/cupertino/picker.dart

```diff
@@ -488,7 +491,21 @@ class _RenderCupertinoPickerSemantics extends RenderProxyBox {
       return;
     }
     _currentIndex = controller.selectedItem;
-    markNeedsSemanticsUpdate();
+    // The semantics of this render object depends on the semantics node in
+    // the subtree. It is not guaranteed the semantics subtree is up-to-date
+    // when [assembleSemanticsNode] is called. Thus, delay a frame to
+    // ensure the subtree has updated.
+    //
+    // TODO(chunhtai): needs a better way to listen to semantics subtree.
```

Comment:
What would be the better way to do this? We should be careful with delaying things by a frame. This often leads to other problems...

Reply:
I don't have anything concrete yet. some ideas i have

- maybe semanticsnode can register change listener, this may lead to circular dependency though
- update semantics node from bottom up, this may lead to performance implication as we may calculate node that get throws a way at the end.
- use childSemanticsDelegate to create incomplete fragment with desired property, in this case, gets the label from child semantics and  replace it with a incompletefragment with value set to the childSemantics.label.

Reply:
actually I think build from bottom up may be ok... let me try it

Reply:
If we really want to do a bottom up, we have to do the semantics update in three pass. 1. update parentdata.(topdown) 2, update geometry.(topdown) 3 builds semantics node(bottom up)